### PR TITLE
Various fixes and some added features

### PR DESCRIPTION
**Pull request by suricactus**
I really enjoy the existence of this plugin! Finally some convenience in this world. I've fixed a number of bugs that I've encountered, and also added some nice to have features.
I hope the commit messages are clear enough and the code good enough to be merged.

### CODE REVIEW

**Comment by kbgg** [comment]
Thank you for your PR! These look like some awesome additions and I'll be happy to merge them.
Just a few things need to be changed:
- Could you close this PR and create it again against the `dev` branch instead? Master is used for the current stable release version of the plugin
- The linter came back with a couple of PEP8 formatting violations:
```
./controllers/results_dialog.py:8:1: F401 'qgis.gui.QgsMapCanvas' imported but unused
./controllers/results_dialog.py:9:80: E501 line too long (88 > 79 characters)
./controllers/results_dialog.py:44:80: E501 line too long (84 > 79 characters)
./controllers/results_dialog.py:203:80: E501 line too long (88 > 79 characters)
./controllers/results_dialog.py:217:80: E501 line too long (100 > 79 characters)
./controllers/results_dialog.py:220:80: E501 line too long (121 > 79 characters)
./controllers/results_dialog.py:226:80: E501 line too long (80 > 79 characters)
./controllers/query_dialog.py:26:80: E501 line too long (106 > 79 characters)
./controllers/query_dialog.py:33:80: E501 line too long (95 > 79 characters)
./controllers/query_dialog.py:34:80: E501 line too long (99 > 79 characters)
```
- One small typo which I made a comment about above

Thank you!

File: views/results_dialog.ui [hunk 1]
```diff
@@ -145,6 +145,19 @@
            </property>
           </spacer>
          </item>
+         <item>
+          <widget class="QPushButton" name="previewExternalButton">
+           <property name="enabled">
+            <bool>false</bool>
+           </property>
+           <property name="toolTip">
+            <string>Opens the preview image in the preffered external image viewere</string>
```
Comment (kbgg) [inline views/results_dialog.ui]: `viewer` not `viewere`

**Comment by suricactus** [comment]
I think I now comply with the three bullets you listed above.
However, there is a conflict with your fix to disable the button, I think mine solution fixes more cases.

IMHO Changing the linter to accept lines that accept at least 120 characters would improve the overall readability of the code, as we are dealing with QT and QGIS interface, which tend to be quite verbose. For example: https://github.com/suricactus/qgis-stac-browser/blob/fixes/controllers/query_dialog.py#L26-L28 .

I will put a few more "controvesial" suggestions in the issue list, I'll be happy to implement them if you agree they are needed. Adding filters is already on my TODO list.

**Comment by kbgg** [comment]
> However, there is a conflict with your fix to disable the button, I think mine solution fixes more cases.

Merged in your fix

> IMHO Changing the linter to accept lines that accept at least 120 characters would improve the overall readability of the code, as we are dealing with QT and QGIS interface, which tend to be quite verbose. For example: https://github.com/suricactus/qgis-stac-browser/blob/fixes/controllers/query_dialog.py#L26-L28 .

Good point, I'll create a ticket for updating the linter to accept lines to 120 characters.

> I will put a few more "controvesial" suggestions in the issue list, I'll be happy to implement them if you agree they are needed. Adding filters is already on my TODO list.

Keep the suggestions coming! The best way to have a great plugin is to work collaboratively

**Comment by suricactus** [comment]
Wohoo, happy to see my PR merged. One more thing I forgot to mention is to check if "View in external image viewer" button works on Windows/Mac. I've checked it on Linux and seems to be fine. According to the documentation, this is the official way to invoke it, but for such API some empyrical confirmation is better.

**Comment by kbgg** [comment]
I checked it on Mac and it works there. We'll have to update some processes for testing on different platforms before we cut a release.

**Pull request merged**